File: csharp/adapter.cs
class ExternalClass {
    public string Search() => "get user info";
}

interface IInternal {
    string Fetch();
}

class Adapter : IInternal {
    private ExternalClass external;

    public Adapter(ExternalClass external) {
        this.external = external;
    }

    public string Fetch() => external.Search();
}

File: assignment_4/c++/adapter.cpp
#include <string>
#include <iostream>

using namespace std;

class ExternalClass {
    public:
        string search() {
            return "get user info";
        }
    };
    
    class Adapter {
        ExternalClass* external;
    public:
        Adapter(ExternalClass* ext) : external(ext) {}
    
        string fetch() {
            return external->search();
        }
    };

int main() {
    ExternalClass* legacyAPI = new ExternalClass();
    Adapter adapter(legacyAPI);

    cout << adapter.fetch() << endl;  

    delete legacyAPI; 
    return 0;
}
